crossfit/backend/torch/hf/model.py

```diff
@@ -81,11 +96,13 @@ def fit_memory_estimate_curve(self, model=None):
         y = []
 
         max_seq = self.max_seq_length()
-        for batch_size in tqdm(range(2048, 0, -256)):
+        for batch_size in tqdm(
+            range(self.end_batch_size, self.start_batch_size, -self.batch_size_increment)
+        ):
             if batch_size <= 0:
                 continue
```

Comment:
Is there a reason you start at the largest batch size first and then go down? In my experience with llama 7B, a lot of time was wasted on the larger batch sizes and larger sequence lengths. It would safe a bit of time if you could stop early. In the inner loop, you could stop when the sequence length is too long given the current batch size and it triggers an OOM. For the outer loop, you could stop when all sequence lengths at the current batch size triggered an OOM.

It's a minor thing compared to the time it takes to run the models on large datasets obviously, but it was annoying when repeatedly debugging with a small dataset.

Reply:
I agree, it makes sense to go from small to large as that allows early stopping.

Reply:
Resolved by: https://github.com/rapidsai/crossfit/pull/66/commits/bbdc4f2dec05cd31168bb8e0b2e9ad62ec586e54